cmake/kokkos_tribits.cmake

```diff
@@ -288,6 +306,43 @@ function(KOKKOS_SET_LIBRARY_PROPERTIES LIBRARY_NAME)
     target_compile_options(
       ${LIBRARY_NAME} PUBLIC $<$<COMPILE_LANGUAGE:${KOKKOS_COMPILE_LANGUAGE}>:${NODEDUP_XCOMPILER_OPTIONS}>
     )
+    list(APPEND ALL_KOKKOS_COMPILER_FLAGS ${NODEDUP_XCOMPILER_OPTIONS})
+  endif()
+
+  message(STATUS "Checking if compiler can consume CMAKE_${KOKKOS_COMPILE_LANGUAGE}_FLAGS and Kokkos compile flags")
```

Comment:
we first check if user provided and our args work, if not we try the two sets individually